### PR TITLE
torch dataloader for librispeech JAX workloads 

### PR DESCRIPTION
**Pull request by sourabh2k15**

### CODE REVIEW

File: algorithmic_efficiency/workloads/librispeech_conformer/workload.py [hunk 1]
```diff
@@ -100,17 +101,36 @@ def _build_dataset(self,
...
-                                                global_batch_size,
-                                                num_batches)
-
-    for batch in iter(ds):
-      batch = data_utils.shard_and_maybe_pad_np(
-          batch, padding_value=1, global_batch_size=global_batch_size)
-      yield batch
+    ds = LibriSpeechDataset(split=split, data_dir=data_dir)
+    ds_iter_batch_size = global_batch_size
+    sampler = None
```
Comment (znado) [inline algorithmic_efficiency/workloads/librispeech_conformer/workload.py:106]: nit: delete and just pass `sampler=None` on L112

File: algorithmic_efficiency/workloads/librispeech_conformer/workload.py [hunk 2]
```diff
@@ -100,17 +101,36 @@ def _build_dataset(self,
...
-                                                train,
-                                                global_batch_size,
-                                                num_batches)
-
-    for batch in iter(ds):
-      batch = data_utils.shard_and_maybe_pad_np(
-          batch, padding_value=1, global_batch_size=global_batch_size)
-      yield batch
+    ds = LibriSpeechDataset(split=split, data_dir=data_dir)
+    ds_iter_batch_size = global_batch_size
```
Comment (znado) [inline algorithmic_efficiency/workloads/librispeech_conformer/workload.py:105]: nit: delete and just pass `batch_size=global_batch_size` on L110

File: algorithmic_efficiency/workloads/librispeech_conformer/workload.py [hunk 3]
```diff
@@ -100,17 +101,36 @@ def _build_dataset(self,
...
+        batch_size=ds_iter_batch_size,
+        shuffle=train,
+        sampler=sampler,
+        num_workers=4,
+        prefetch_factor=10,
+        pin_memory=False,
+        drop_last=train,
+    )
+
+    dataloader = data_utils.cycle(
```
Comment (znado) [inline algorithmic_efficiency/workloads/librispeech_conformer/workload.py:119]: given that the kwargs default to false, can you just make this `data_utils.cycle(dataloader)` ?

**Review by znado — CHANGES_REQUESTED** [review]
Nice! Two comments:
- can we merge the `_build_dataset` into `_build_input_queue`? it doesn't look like it's actually called anywhere
- we should delete the current `algorithmic_efficiency/workloads/librispeech_conformer/input_pipeline.py` and move the pytorch one to be there (and fix imports to it) so that we don't have an unused data loader laying around (if we need the code in the future we can always go back in git history)

File: algorithmic_efficiency/workloads/librispeech_conformer/workload.py (hunk 2 above)
Comment (sourabh2k15) [reply algorithmic_efficiency/workloads/librispeech_conformer/workload.py:105]: done

File: algorithmic_efficiency/workloads/librispeech_conformer/workload.py (hunk 1 above)
Comment (sourabh2k15) [reply algorithmic_efficiency/workloads/librispeech_conformer/workload.py:106]: done

**Review by znado — APPROVED** [review]

**Pull request merged**